pennylane/ops/op_math/exp.py

```diff
@@ -249,15 +273,18 @@ def matrix(self, wire_order=None):
                     "Use a different interface if you need backpropagation.",
                     UserWarning,
                 )
-        base_mat = (
-            qml.matrix(self.base) if isinstance(self.base, qml.Hamiltonian) else self.base.matrix()
-        )
-        if coeff_interface == "torch":
-            # other wise get `RuntimeError: Can't call numpy() on Tensor that requires grad.`
-            base_mat = math.convert_like(base_mat, self.coeff)
-        mat = math.expm(self.coeff * base_mat)
+        return super().matrix(wire_order=wire_order)
 
-        return expand_matrix(mat, wires=self.wires, wire_order=wire_order)
+    @staticmethod
+    def _matrix(scalar, mat):
+        scalar_interface = math.get_interface(scalar)
+        if scalar_interface == "torch":
+            # other wise get `RuntimeError: Can't call numpy() on Tensor that requires grad.`
+            mat = math.convert_like(mat, scalar)
+        elif scalar_interface == "tensorflow" and not math.iscomplex(scalar):
+            # cast scalar to complex to avoid an error
+            scalar = math.cast_like(scalar, mat)
```

Comment:
We definitely should implement this functionality somewhere central! It is required by virtually _all_ `parametric_ops` `compute_matrix` methods.

Reply:
Moved in `ScalarSymbolicOp.matrix` method!

Reply:
If we want to generalise it even further I'd recommend doing it in a separate PR

Reply:
I agree, it's just something I've seen so often, and usually there is no case separation done. I'll start a discussion on this.